### ISSUE TITLE
Grails 3 Support

### ISSUE THREAD

**Issue by ctoestreich** [splix/grails-spring-security-facebook#91]
Support for new Spring Security 3.x and Grails 3.x

**Comment by dinizssa** [comment]
Is there any ETA for this? I'm wondering if we could start using this plugin for Grails 3.1.1 in near future.

**Comment by splix** [comment]
I guess it's not accepted yet by main Grails repository yet. But you can try to add following repository as a workaround:
```
repositories {
    maven {
        url  "http://dl.bintray.com/splix/grails-plugins" 
    }
}
```

**Comment by dinizssa** [comment]
Thanks, @splix. It works!

**Comment by dinizssa** [comment]
Hi Igor,

Do you know why this below?

Caused by: org.springframework.beans.factory.BeanCreationException: Error
creating bean with name 'facebookAuthRedirectFilter': Cannot resolve
reference to bean 'redirectFailureHandlerExample' while setting bean
property 'authenticationFailureHandler'; nested exception is
org.springframework.beans.factory.NoSuchBeanDefinitionException: No bean
named 'redirectFailureHandlerExample' is defined
at
org.springframework.beans.factory.support.BeanDefinitionValueResolver.resolveReference(BeanDefinitionValueResolver.java:359)
at
org.springframework.beans.factory.support.BeanDefinitionValueResolver.resolveValueIfNecessary(BeanDefinitionValueResolver.java:108)
at
org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyPropertyValues(AbstractAutowireCapableBeanFactory.java:1475)
at
org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1220)
at
org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:537)
at
org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:476)
at
org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:303)
at
org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
at
org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:299)
at
org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)
at
org.springframework.boot.context.embedded.ServletContextInitializerBeans.getOrderedBeansOfType(ServletContextInitializerBeans.java:211)
at
org.springframework.boot.context.embedded.ServletContextInitializerBeans.addAsRegistrationBean(ServletContextInitializerBeans.java:167)
at
org.springframework.boot.context.embedded.ServletContextInitializerBeans.addAsRegistrationBean(ServletContextInitializerBeans.java:162)
at
org.springframework.boot.context.embedded.ServletContextInitializerBeans.addAdaptableBeans(ServletContextInitializerBeans.java:144)
at
org.springframework.boot.context.embedded.ServletContextInitializerBeans.<init>(ServletContextInitializerBeans.java:74)
at
org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.getServletContextInitializerBeans(EmbeddedWebApplicationContext.java:233)
at
org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.selfInitialize(EmbeddedWebApplicationContext.java:220)
at
org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.access$000(EmbeddedWebApplicationContext.java:84)
at
org.springframework.boot.context.embedded.EmbeddedWebApplicationContext$1.onStartup(EmbeddedWebApplicationContext.java:205)
at
org.springframework.boot.context.embedded.tomcat.TomcatStarter.onStartup(TomcatStarter.java:54)
at
org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5170)
at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
at
org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1408)
at
org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1398)
at java.util.concurrent.FutureTask.run(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at java.lang.Thread.run(Unknown Source)
Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException:
No bean named 'redirectFailureHandlerExample' is defined
at
org.springframework.beans.factory.support.DefaultListableBeanFactory.getBeanDefinition(DefaultListableBeanFactory.java:687)
at
org.springframework.beans.factory.support.AbstractBeanFactory.getMergedLocalBeanDefinition(AbstractBeanFactory.java:1168)
at
org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:281)
at
org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:194)
at
org.springframework.beans.factory.support.BeanDefinitionValueResolver.resolveReference(BeanDefinitionValueResolver.java:351)
... 27 more

I'm trying to run you example using Grails 3.1.1, I can see the beans
defined at resources.groovy, but it seems Grails is not able to find them.

Thanks,

Andre.

On 8 March 2016 at 10:11, Igor Artamonov <notifications@github.com> wrote:

> Closed #91
> <https://github.com/splix/grails-spring-security-facebook/issues/91>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/splix/grails-spring-security-facebook/issues/91#event-581700359>
> .
>



-- 
André Diniz
Twitter: @dinizssa

**Comment by dinizssa** [comment]
Just figured out now:

Thing is defaultFailureUrl is now private so we should do this below:

redirectFailureHandlerTest(SimpleUrlAuthenticationFailureHandler) {
*  setDefaultFailureUrl('/login/auth') //redirect to this URL when
authentication fails*
}

You might need to update your plugin later. Nice plugin by the way!

Cheers,

Andre.

On 8 March 2016 at 16:35, André Diniz <adiniz@gmail.com> wrote:

> Hi Igor,
>
> Do you know why this below?
>
> Caused by: org.springframework.beans.factory.BeanCreationException: Error
> creating bean with name 'facebookAuthRedirectFilter': Cannot resolve
> reference to bean 'redirectFailureHandlerExample' while setting bean
> property 'authenticationFailureHandler'; nested exception is
> org.springframework.beans.factory.NoSuchBeanDefinitionException: No bean
> named 'redirectFailureHandlerExample' is defined
> at
> org.springframework.beans.factory.support.BeanDefinitionValueResolver.resolveReference(BeanDefinitionValueResolver.java:359)
> at
> org.springframework.beans.factory.support.BeanDefinitionValueResolver.resolveValueIfNecessary(BeanDefinitionValueResolver.java:108)
> at
> org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyPropertyValues(AbstractAutowireCapableBeanFactory.java:1475)
> at
> org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1220)
> at
> org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:537)
> at
> org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:476)
> at
> org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:303)
> at
> org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
> at
> org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:299)
> at
> org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)
> at
> org.springframework.boot.context.embedded.ServletContextInitializerBeans.getOrderedBeansOfType(ServletContextInitializerBeans.java:211)
> at
> org.springframework.boot.context.embedded.ServletContextInitializerBeans.addAsRegistrationBean(ServletContextInitializerBeans.java:167)
> at
> org.springframework.boot.context.embedded.ServletContextInitializerBeans.addAsRegistrationBean(ServletContextInitializerBeans.java:162)
> at
> org.springframework.boot.context.embedded.ServletContextInitializerBeans.addAdaptableBeans(ServletContextInitializerBeans.java:144)
> at
> org.springframework.boot.context.embedded.ServletContextInitializerBeans.<init>(ServletContextInitializerBeans.java:74)
> at
> org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.getServletContextInitializerBeans(EmbeddedWebApplicationContext.java:233)
> at
> org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.selfInitialize(EmbeddedWebApplicationContext.java:220)
> at
> org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.access$000(EmbeddedWebApplicationContext.java:84)
> at
> org.springframework.boot.context.embedded.EmbeddedWebApplicationContext$1.onStartup(EmbeddedWebApplicationContext.java:205)
> at
> org.springframework.boot.context.embedded.tomcat.TomcatStarter.onStartup(TomcatStarter.java:54)
> at
> org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5170)
> at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
> at
> org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1408)
> at
> org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1398)
> at java.util.concurrent.FutureTask.run(Unknown Source)
> at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
> at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
> at java.lang.Thread.run(Unknown Source)
> Caused by:
> org.springframework.beans.factory.NoSuchBeanDefinitionException: No bean
> named 'redirectFailureHandlerExample' is defined
> at
> org.springframework.beans.factory.support.DefaultListableBeanFactory.getBeanDefinition(DefaultListableBeanFactory.java:687)
> at
> org.springframework.beans.factory.support.AbstractBeanFactory.getMergedLocalBeanDefinition(AbstractBeanFactory.java:1168)
> at
> org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:281)
> at
> org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:194)
> at
> org.springframework.beans.factory.support.BeanDefinitionValueResolver.resolveReference(BeanDefinitionValueResolver.java:351)
> ... 27 more
>
> I'm trying to run you example using Grails 3.1.1, I can see the beans
> defined at resources.groovy, but it seems Grails is not able to find them.
>
> Thanks,
>
> Andre.
>
> On 8 March 2016 at 10:11, Igor Artamonov <notifications@github.com> wrote:
>
>> Closed #91
>> <https://github.com/splix/grails-spring-security-facebook/issues/91>.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/splix/grails-spring-security-facebook/issues/91#event-581700359>
>> .
>>
>
>
>
> --
> André Diniz
> Twitter: @dinizssa
>
>


-- 
André Diniz
Twitter: @dinizssa
